### ISSUE TITLE
problem with IIC2 

### ISSUE THREAD

**Issue by poojaasritha** [rogerclarkmelbourne/Arduino_STM32#613]
error while uncommenting the below line in i2c scanner example....
please fix it asap.................
and also show us a way on how to use I2C2 LINE with Wire_slave.h
//TwoWire WIRE2 (2,I2C_FAST_MODE);

error that i'm getting is :
'WIRE2' was not declared in this scope

**Comment by DeadSHoT555** [comment]
I also face same problem.
Please show us a quick solution to use that

**Comment by rajjjjjh** [comment]
Yeah i too can't communicate with second  i2c line...  How to communicate on second i2c line??????

**Comment by GitMoDu** [comment]
`#include <Wire_slave.h> `

Replace with 

`#include <Wire.h> `

Now you can instantiate your prefered I2C channel.

```
TwoWire WireInstance(2,I2C_FAST_MODE);

WireInstance.beginTransmission(adderss)
```

**Comment by rikykumar** [comment]
in that case the error was
  Wire.onRequest and Wire.onRecieve were not declared

**Comment by rikykumar** [comment]
could not use stm32f103c as slave with <Wire.h>

**Comment by rikykumar** [comment]
do u have any option to use stm32 as slave on Wire2

**Comment by GitMoDu** [comment]
I2C slave is not supported on STM32 duino, on either channel.

**Comment by rikykumar** [comment]
no.., using <Wire_slave.h> it is possible to communicate as slave with stm32 ..., i have tested it .., but i dont know how to use ion on second wire

**Comment by rikykumar** [comment]
there are direct inbuild examples with package to user stm32 as slave under "Wire_slave" folder

**Comment by LMESTM** [comment]
I think you can use other instances of I2C (there can be more than 2) by using an alternative, declaration with SCL and SDA pins:     TwoWire(uint8_t sda, uint8_t scl);

Like:
TwoWire Wire2(PB_11, PB_10);

Wouldn't that unblock the situation ?

**Comment by GitMoDu** [comment]
You're correct, it compiles at least. As far as I knew, I2C support wasn't there for slave. I don't how much more I can help you.

**Comment by DeadSHoT555** [comment]
![IMG_20190402_225612_183](https://user-images.githubusercontent.com/39310462/55423009-a26eb680-559a-11e9-9636-39e3f1081284.jpg)
Couldn't get it

**Comment by DeadSHoT555** [comment]
![IMG_20190402_225615_712](https://user-images.githubusercontent.com/39310462/55423022-ad294b80-559a-11e9-9df6-0e873ff0d51f.jpg)
And while using <Wire.h> could'nt work out slave mode but can hanle Wire2

**Comment by stevstrong** [comment]
Possible duplicate of #637, #638 .

**Comment by stevstrong** [comment]
Needs implementing an example to simultaneously use Wire as master and slave.
Possible duplicate of #625
